File: testing/MyUnitTest/java/app/src/main/java/com/dicoding/picodiploma/myunittest/MainViewModel.java
package com.dicoding.picodiploma.myunittest;

import androidx.lifecycle.ViewModel;

class MainViewModel extends ViewModel {
    private final CuboidModel cuboidModel;

    MainViewModel(CuboidModel cuboidModel) {
        this.cuboidModel = cuboidModel;
    }

    void save(double l, double w, double h) {
        cuboidModel.save(l, w, h);
    }

    double getCircumference() {
        return cuboidModel.getCircumference();
    }

    double getSurfaceArea() {
        return cuboidModel.getSurfaceArea();
    }

    double getVolume() {
        return cuboidModel.getVolume();
    }
}

File: testing/LatihanUITest/kotlin/app/src/main/java/com/dicoding/picodiploma/myunittest/MainViewModel.kt
package com.dicoding.picodiploma.myunittest

internal class MainViewModel(private val cuboidModel: CuboidModel) {

    fun getCircumference() = cuboidModel.getCircumference()

    fun getSurfaceArea() = cuboidModel.getSurfaceArea()

    fun getVolume() = cuboidModel.getVolume()

    fun save(l: Double, w: Double, h: Double) {
        cuboidModel.save(l, w, h)
    }
}
